### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by Bleu-Yves-Sopoude**
In this feature we:

- Added Redux Toolkit
- Created a directory that will contain all Redux logic
- Configured a Redux store 
- Defined slices of state for book and categories

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export const BookSlice = createSlice({
+  name: "books",
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book.id !== action.payload);
```
Comment (danmainah) [inline src/redux/books/bookSlice.js]: - [ ] @Bleu-Yves-Sopoude Great work on your reducer, however, the removeBook action is suppose to check if the book is  not equal to action.id, you can use the example below to ensure the action will work correctly:
```suggestion
      addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },

```

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const initialState = {
+  categories: [],
+  status: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    categoriesStatus: (state) => state.status,
```
Comment (danmainah) [inline src/redux/categories/categoriesSlice.js]: - [ ] @Bleu-Yves-Sopoude Great work on your categories reducer, however, the categoriesStatus action is supposed to check the `Under construction` status.
```suggestion
checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
```

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Bleu-Yves-Sopoude , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Implemented adding the redux directory correctly following the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
✔️ Added the redux store correctly.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Bleu-Yves-Sopoude, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**